src/app/customer/checkoutSuggestion/BoltCheckoutSuggestion.tsx

```diff
@@ -0,0 +1,94 @@
+import { CheckoutSelectors, CustomerInitializeOptions, CustomerRequestOptions, ExecutePaymentProviderCheckoutOptions } from '@bigcommerce/checkout-sdk';
+import { noop } from 'lodash';
+import React, { memo, useEffect, useState, FunctionComponent, SyntheticEvent } from 'react';
+
+import { TranslatedString } from '../../locale';
+import { Button, ButtonVariant } from '../../ui/button';
+import { IconBolt } from '../../ui/icon';
+
+export interface BoltCheckoutSuggestionProps {
+    isExecutingPaymentProviderCheckout: boolean;
+    methodId: string;
+    deinitializeCustomer(options: CustomerRequestOptions): Promise<CheckoutSelectors>;
+    executePaymentProviderCheckout(options: ExecutePaymentProviderCheckoutOptions): Promise<CheckoutSelectors>;
+    initializeCustomer(options: CustomerInitializeOptions): Promise<CheckoutSelectors>;
+    onUnhandledError?(error: Error): void;
+}
+
+const BoltCheckoutSuggestion: FunctionComponent<BoltCheckoutSuggestionProps> = ({
+    isExecutingPaymentProviderCheckout,
+    methodId,
+    deinitializeCustomer,
+    executePaymentProviderCheckout,
+    initializeCustomer,
+    onUnhandledError = noop,
+}) => {
+    const [ showSuggestion, setShowSuggestion ] = useState<boolean>(false);
+
+    useEffect(() => {
+        deinitializeCustomer({ methodId });
+
+        try {
+            initializeCustomer({
+                methodId,
+                [methodId]: {
+                    onInit: (hasBoltAccount: boolean) => {
```

Comment:
You can omit `boolean`.

Reply:
I can't remove `boolean` type here because of tslint error

Reply:
Ah I think that's because TypeScript doesn't know `methodId` is `bolt`. If it doesn't know, it won't know how to infer to `BoltCustomerInitializeOptions` interface. If you write `bolt` instead of `methodId` or check that `methodId === 'bolt'`, you shouldn't have to write `boolean`. `BoltCheckoutSuggestion` is specifically for Bolt so you probably don't need to reference `methodId`.

Reply:
Oh, I see. Ok, I will fix it